collector/metrics/metric_collector.go

```diff
@@ -67,22 +63,26 @@ func NewMetricCollector(
 	dataSource int,
 	detectInterval time.Duration,
 	detectTimeout time.Duration) MetricCollector {
-	DataSource = dataSource
 	GaugeMetricsMap = make(map[string]prometheus.GaugeVec, 128)
 	CounterMetricsMap = make(map[string]prometheus.CounterVec, 128)
 	SummaryMetricsMap = make(map[string]prometheus.Summary, 128)
-	RoleByDataSource = make(map[int]string, 128)
 	TableNameByID = make(map[string]string, 128)
-	RoleByDataSource[0] = "meta"
-	RoleByDataSource[1] = "replica"
-	initMetrics()
 
-	return &Collector{detectInterval: detectInterval, detectTimeout: detectTimeout}
+	var role string
+	if dataSource == 0 {
```

Comment:
We could just change `dataSource` to `role`, and certainly change the type of `dataSource` to `string` instead of `int` which means, changing the type of following constants to `string`:
```go
const (
        MetaServer    int = 0
        ReplicaServer int = 1
)
```